### ISSUE TITLE
As a multiple time failure at NaNoWriMo.... I'm in!

### ISSUE THREAD

**Issue by darkliquid** [dariusk/NaNoGenMo#9]
I've got some big, big ideas that will probably fall to the wayside in favour of quick and dirty hacks, but I'm in.

I'm hoping I'll end up coming up with something better than just doing this: 

```
curl -O http://www.gutenberg.org/cache/epub/23641/pg23641.txt
```

**Comment by darkliquid** [comment]
Well, I've started using the dada engine with a bunch of collected words from various lists and my extracted sentence structures to start generating titles for the novel. So far, I've got some choice titles such as:
- Immaculate passenger Sherill Nada Watkins
- Breads F'thagn
- Parties offer Lanny Aaron Bacon branch shirt nerve behavior
- Cristobal Gleason 58 each giants Reginia Cochran-Cordero dolls Daren Dill
- Held spells cautious even if 171 wastes Isaura Kauffman-Holland

I've not yet done anything with punctuation, so that needs some tweaking, but otherwise those have some semblance of sense to them.

Feel free to use my dada engine include files to help with generating your own names, etc.

**Comment by darkliquid** [comment]
First novel posted. Genrated using the dada engine using grammar derived form my various analysis. It was a massive chore to get it to work at all as the dada engine kept 'forgetting' rule definitions randomly.

https://github.com/darkliquid/NaNoGenMo/blob/master/novels/first-novel.txt.gz

**Comment by dariusk** [comment]
FYI that link doesn't give me a valid gzip file, folks should use this one instead:

https://github.com/darkliquid/NaNoGenMo/blob/master/novels/first-novel.txt.gz?raw=true

**Comment by darkliquid** [comment]
Ahh thanks for that.

I've been playing with the dada-engine for most of the actual generation from my various extracted info sources, but either I'm doing something wrong or it's struggling with some of the more complex and very large grammars I'm throwing at it so I'm probably going to drop it for some custom stuff for my next attempt. For example, my dada grammar includes about 200k-300k individual entries for words, names, cities, etc in addition to almost 100k rules regarding sentence structures, which makes for quite a large thing for it to work with. At random it decides that one of the previously defined rules is no longer defined. This could be a fault with how I've written my grammar (I generated a lot of it automatically using python scripts) or a fault in dada-engine, but I don't wont to waste time debugging this when I have other ideas to try out.

**TL;DR** dada-engine might struggle with large, complex grammars, or I may suck, so switching tactics.
